engine/src/main/java/org/terasology/logic/console/ui/MiniChatOverlay.java

```diff
@@ -62,13 +65,21 @@ public void initialise() {
             @Override
             public String get() {
                 Iterable<Message> msgs = console.getMessages(CoreMessageType.CHAT, CoreMessageType.NOTIFICATION);
-                String last = "";
+                ArrayList<Message> msgsList = new ArrayList<>();
+                StringBuilder messageHistory = new StringBuilder();
 
                 for (Message msg : msgs) {
-                    last = msg.getMessage();
+                    msgsList.add(msg);
                 }
 
-                return last;
+                //checks if number of messages > MAX_LINES, else starts from 0
+                for (int i = msgsList.size() >= MAX_LINES ? msgsList.size() - MAX_LINES : 0; i < msgsList.size(); i++) {
+                    messageHistory.append(msgsList.get(i).getMessage());
+                    if (i < msgsList.size() - 1) {
+                        messageHistory.append(Console.NEW_LINE);
+                    }
+                }
+                return messageHistory.toString();
```

Comment:
Checked out and worked well so far.
Small experiment on readability (not sure what to prefer, I would leave this up to you):
```
        message.bindText(new ReadOnlyBinding<String>() {
            @Override
            public String get() {
                Iterable<Message> messages = console.getMessages(CoreMessageType.CHAT, CoreMessageType.NOTIFICATION);
                int messagesCount = Iterables.size(messages);
                int messagesToSkip = messagesCount - MAX_LINES;
                if (messagesToSkip > 0) {
                    messages = Iterables.skip(messages, messagesToSkip);
                }

                Stream<Message> mostRecentMessages = StreamSupport.stream(messages.spliterator(), false);
                String messagesHistory = mostRecentMessages.map(Message::getMessage).collect(Collectors.joining(Console.NEW_LINE));
                return messagesHistory;
            }
        });
```

In any case, `MAX_LINES` should probably be `MAX_MESSAGES` or `MAX_VISIBLE_MESSAGES`. Fun fact: The overlay can get really huge, when large messages wrap over multiple lines (tested with large `BANANA...AS` 👓 . However adding a scrollpane or a real line limitiation/char is likely out of scope for this PR, so this is fine from my side :)